diff --git a/SConstruct b/SConstruct
index 1092d82..35c19a9 100644
--- a/SConstruct
+++ b/SConstruct
@@ -235,7 +235,6 @@ elif env["platform"] == "ios":
     if env["ios_simulator"]:
         sdk_name = "iphonesimulator"
         env.Append(CCFLAGS=["-mios-simulator-version-min=10.0"])
-        env["LIBSUFFIX"] = ".simulator" + env["LIBSUFFIX"]
     else:
         sdk_name = "iphoneos"
         env.Append(CCFLAGS=["-miphoneos-version-min=10.0"])
@@ -485,6 +484,8 @@ if env["platform"] == "android":
     arch_suffix = env["android_arch"]
 elif env["platform"] == "ios":
     arch_suffix = env["ios_arch"]
+    if env["ios_simulator"]:
+        arch_suffix += ".simulator"
 elif env["platform"] == "javascript":
     arch_suffix = "wasm"
 elif env["platform"] == "osx":
